scripts/mbedtls_framework/test_case.py

```diff
@@ -89,3 +111,28 @@ def write_data_file(filename: str,
             tc.write(out)
         out.write('\n# End of automatically generated file.\n')
     os.replace(tempfile, filename)
+
+def psa_or_3_6_feature_macro(psa_alg: str,
```

Comment:
I think we need to better define what the `psa_alg` input is allowed to be. My thinking when I suggested this function was it would be similar to `psa_want_symbol`, ie it takes `PSA_ALG_xxx` or `PSA_KEY_TYPE_xxx` as input and returns the corresponding feature macro.

Here his function seems to accept both `PSA_WANT_xxx` and `PSA_ALG_xxx` as input. My first reaction is this looks like unnecessary complexity. However, is there a reason I'm missing?

Minor: since the function accepts not only algorithms but also key types, I don't think `psa_alg` is a great name. Perhaps `psa_name` (similarly to `psa_want_symbol()` that calls its argument `name`, but here we add `psa_` to distinguish from legacy). Or `psa_feature`?

Anyway, the docstring should clarify what kind of inputs are accepted.

Reply:
I did this because while the issue description talked a lot about hashes, it didn't say anything about how to deal with PK symbols. And because I thought that we're also keeping `PSA_HAVE_ALG_SOME_ECDSA` and others that don't follow the same pattern as `PSA_ALG_xxx` or `PSA_KEY_TYPE_xxx`, I decided to cover both options for hashes knowing that one will probably be removed.

Reply:
Ok, makes complete sense. However, I happen to be on a personal crusade against complexity these days, so I'm still inclined to go for the simpler solution :)

Reply:
Totally agree, I'm still in the middle of reworking the PR. Just wanted to share why I did it 😄